src/iam/server/iamserver.cpp

```diff
@@ -0,0 +1,784 @@
+}
+
+void IAMServer::Close()
+{
+    mPublicServer->Shutdown();
+    mProtectedServer->Shutdown();
+
+    mPublicServer->Wait();
+    mProtectedServer->Wait();
+}
+
+/***********************************************************************************************************************
+ * IAMPublicService implementation
+ **********************************************************************************************************************/
+
+grpc::Status IAMServer::GetAPIVersion(
+    grpc::ServerContext* context, const google::protobuf::Empty* request, iamanager::v4::APIVersion* response)
+{
+    (void)context;
+    (void)request;
+
+    LOG_DBG() << "Process get API version";
+
+    response->set_version(cIamAPIVersion);
+
+    return grpc::Status::OK;
+}
+
+grpc::Status IAMServer::GetNodeInfo(
+    grpc::ServerContext* context, const google::protobuf::Empty* request, iamanager::v4::NodeInfo* response)
+{
+    (void)context;
+    (void)request;
+
+    LOG_DBG() << "Process get node info";
+
+    response->set_node_id(mNodeID);
+    response->set_node_type(mNodeType);
+
+    return grpc::Status::OK;
+}
+
+grpc::Status IAMServer::GetCert(grpc::ServerContext* context, const iamanager::v4::GetCertRequest* request,
+    iamanager::v4::GetCertResponse* response)
+{
+    (void)context;
+
+    LOG_DBG() << "Process get cert request: type=" << request->type().c_str()
+              << ", serial=" << request->serial().c_str();
+
+    response->set_type(request->type());
+
+    auto issuer = ConvertByteArrayToAOS(request->issuer());
+
+    StaticArray<uint8_t, crypto::cSerialNumSize> serial;
+
+    auto err = String(request->serial().c_str()).HexToByteArray(serial);
+    if (!err.IsNone()) {
+        LOG_ERR() << "Serial conversion failed: " << err;
+
+        return grpc::Status(grpc::StatusCode::INTERNAL, "Serial conversion failed");
+    }
+
+    certhandler::CertInfo certInfo;
+
+    err = mCertHandler->GetCertificate(request->type().c_str(), issuer, serial, certInfo);
+    if (!err.IsNone()) {
+        LOG_ERR() << "Get certificate error: " << err;
+
+        return grpc::Status(grpc::StatusCode::INTERNAL, "Get certificate error");
+    }
+
+    response->set_key_url(certInfo.mKeyURL.CStr());
+    response->set_cert_url(certInfo.mCertURL.CStr());
+
+    return grpc::Status::OK;
+}
+
+/***********************************************************************************************************************
+ * IAMPublicIdentityService implementation
+ **********************************************************************************************************************/
+
+grpc::Status IAMServer::GetSystemInfo(
+    grpc::ServerContext* context, const google::protobuf::Empty* request, iamanager::v4::SystemInfo* response)
+{
+    (void)context;
+    (void)request;
+
+    LOG_DBG() << "Process get system info";
+
+    auto [systemID, err1] = mIdentHandler->GetSystemID();
```

Comment:
`err1`, `err2`, `err...` looks not so good. Reuse one `err` var.

Reply:
fixed